File: Circular_Linked_list/circular_doubly_linked_list_create_n_nodes.py
# File Name: circular_doubly_linked_list_create_n_nodes.py
# Program to create a circular doubly linked list with n nodes

class Node:
    def __init__(self, data):
        self.data = data
        self.prev = None
        self.next = None

head = None

# Function to create circular doubly linked list with n nodes
def create_list(n):
    global head
    temp = None

    for i in range(1, n + 1):
        value = int(input(f"Enter data for node {i}: "))
        new_node = Node(value)

        if head is None:
            head = new_node
            head.next = head
            head.prev = head
            temp = head
        else:
            new_node.next = head
            new_node.prev = temp
            temp.next = new_node
            head.prev = new_node
            temp = new_node

if __name__ == "__main__":
    n = int(input("Enter number of nodes: "))
    create_list(n)

File: Circular_Linked_list/CircularDoublyLinkedListCreateNNodes.java
// File Name: CircularDoublyLinkedListCreateNNodes.java
// Program to create a circular doubly linked list with n nodes

import java.util.Scanner;

class Node {
    int data;
    Node prev;
    Node next;

    Node(int data) {
        this.data = data;
        this.prev = null;
        this.next = null;
    }
}

public class CircularDoublyLinkedListCreateNNodes {
    static Node head = null;

    // Function to create circular doubly linked list with n nodes
    static void createList(int n) {
        Scanner sc = new Scanner(System.in);
        Node temp = null;

        for (int i = 1; i <= n; i++) {
            System.out.print("Enter data for node " + i + ": ");
            int value = sc.nextInt();
            Node newNode = new Node(value);

            if (head == null) {
                head = newNode;
                head.next = head;
                head.prev = head;
                temp = head;
            } else {
                newNode.next = head;
                newNode.prev = temp;
                temp.next = newNode;
                head.prev = newNode;
                temp = newNode;
            }
        }
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter number of nodes: ");
        int n = sc.nextInt();

        createList(n);
    }
}
